vendor/sigs.k8s.io/kustomize/api/types/kustomization.go

```diff
@@ -88,6 +93,7 @@ type Kustomization struct {
 	// specification. This can also be done with a patch.
 	Replicas []Replica `json:"replicas,omitempty" yaml:"replicas,omitempty"`
 
+	// Deprecated: Vars will be removed in future release. Migrate to Replacements instead.
```

Comment:
removal of supported functionality from existing versioned kustomize API files is unexpected - kubectl is expected to continue working with existing manifests on upgrade